src/port1.0/portlicensecheck.tcl

```diff
@@ -0,0 +1,276 @@
+#
+
+package provide portlicensecheck 1.0
+package require portutil 1.0
+
+set org.macports.licensecheck [target_new org.macports.licensecheck portlicensecheck::licensecheck_main]
+target_runtype ${org.macports.licensecheck} always
+target_state ${org.macports.licensecheck} no
+target_provides ${org.macports.licensecheck} licensecheck
+target_requires ${org.macports.licensecheck} main
+target_prerun ${org.macports.licensecheck} portlicensecheck::licensecheck_start
+
+namespace eval portlicensecheck {
+}
+
+set_ui_prefix
+
+set check_deptypes [list depends_build depends_lib]
+
+proc portlicensecheck::all_licenses_except { args } {
+    set remaining $licenses::good_licenses
+    foreach arg $args {
+        set remaining [lsearch -inline -all -not -exact $remaining $arg]
+    }
+    return $remaining
+}
+
+# return deps and license for given port
+proc portlicensecheck::infoForPort {portName variantInfo} {
+    set portSearchResult [mport_lookup $portName]
+    if {[llength $portSearchResult] < 1} {
+        puts stderr "Warning: port \"$portName\" not found"
+        return {}
+    }
+    array set portInfo [lindex $portSearchResult 1]
+    set portfile_path [getportdir $portInfo(porturl)]/Portfile
+    set variant_string $variantInfo
+
+    set dependencyList {}
+    set mport [mport_open $portInfo(porturl) [list subport $portInfo(name)] $variantInfo]
+    array unset portInfo
+    array set portInfo [mport_info $mport]
+    # Closing the mport is actually fairly expensive and not really necessary
+    mport_close $mport
+
+    foreach dependencyType $::check_deptypes {
+        if {[info exists portInfo($dependencyType)] && $portInfo($dependencyType) ne ""} {
+            foreach dependency $portInfo($dependencyType) {
+                lappend dependencyList [string range $dependency [string last ":" $dependency]+1 end]
+            }
+        }
+    }
+
+    set ret [list $dependencyList $portInfo(license)]
+    if {[info exists portInfo(installs_libs)]} {
+        lappend ret $portInfo(installs_libs)
+    } else {
+        # when in doubt, assume code from the dep is incorporated
+        lappend ret yes
+    }
+    if {[info exists portInfo(license_noconflict)]} {
+        lappend ret $portInfo(license_noconflict)
+    }
+
+    return $ret
+}
+
+# return license with any trailing dash followed by a number and/or plus sign removed
+set remove_version_re {[0-9.+]+}
+proc portlicensecheck::remove_version {license} {
+    set dash [string last - $license]
+    if {$dash != -1 && [regexp $::remove_version_re [string range $license $dash+1 end]]} {
+        return [string range $license 0 $dash-1]
+    } else {
+        return $license
+    }
+}
+
+proc portlicensecheck::licensecheck_start {args} {
+    global UI_PREFIX subport
+    ui_notice "$UI_PREFIX [format [msgcat::mc "Checking license for %s"] ${subport}]"
+}
+
+proc portlicensecheck::licensecheck_main {args} {
+    # Obtain licenses from '_resources/port1.0/licenses/licenses.tcl'.
```

Comment:
NOTE: As a starting point, I named the new resource area `licenses`, to be more general. With the idea being that the data might be used elsewhere at some point.

But if folks prefer that it be specific to this target, it can be renamed to `licensecheck`.